### PR TITLE
Change V2.0 coors

### PR DESCRIPTION
**Pull request by ZwwWayne**
This PR changes the codebase's coordinate system and has effects on all the operations related to anchors, boxes, and masks.

### CODE REVIEW

File: mmdet/core/anchor/anchor_generator.py [hunk 1]
```diff
@@ -30,8 +30,8 @@ def gen_base_anchors(self):
         w = self.base_size
         h = self.base_size
         if self.ctr is None:
-            x_ctr = 0.5 * (w - 1)
-            y_ctr = 0.5 * (h - 1)
+            x_ctr = 0.
```
Comment (xvjiarui) [inline mmdet/core/anchor/anchor_generator.py:33]:
```
x_ctr = 0.5 * w
y_ctr = 0.5 * h
```

Comment (ZwwWayne) [reply mmdet/core/anchor/anchor_generator.py:33]: The x_ctr should be set as zero.

File: mmdet/core/mask/structures.py [hunk 2]
```diff
@@ -216,18 +220,23 @@ def crop_and_resize(self,
...
+            bboxes = torch.from_numpy(bboxes).to(device=device)
+        if isinstance(inds, np.ndarray):
+            inds = torch.from_numpy(inds).to(device=device)
+
+        num_bbox = bboxes.shape[0]
+        fake_inds = torch.arange(
+            num_bbox, device=device).to(dtype=bboxes.dtype)[:, None]
+        rois = torch.cat([fake_inds, bboxes], dim=1)  # Nx5
+        rois = rois.to(device=device)
+        if num_bbox > 0:
```
Comment (xvjiarui) [inline mmdet/core/mask/structures.py:234]: How about `num_bbox ==0 `?

File: mmdet/models/mask_heads/fcn_mask_head.py [hunk 3]
```diff
@@ -175,34 +180,124 @@ def get_seg_masks(self, mask_pred, det_bboxes, det_labels, rcnn_test_cfg,
...
+    Args:
+        masks: N, 1, H, W
+        boxes: N, 4
+        img_h, img_w (int):
+        skip_empty (bool): only paste masks within the region that
+            tightly bound all boxes, and returns the results this region only.
+            An important optimization for CPU.
+
+    This implementation is modified from
+    https://github.com/facebookresearch/detectron2/
```
Comment (hellock) [inline mmdet/models/mask_heads/fcn_mask_head.py:252]: - Move the description above args
- Add a summary

File: mmdet/models/mask_heads/fcn_mask_head.py [hunk 4]
```diff
@@ -175,34 +180,124 @@ def get_seg_masks(self, mask_pred, det_bboxes, det_labels, rcnn_test_cfg,
...
+
+    Returns:
```
Comment (hellock) [inline mmdet/models/mask_heads/fcn_mask_head.py:254]: Indicate the return type first.

File: mmdet/ops/nms/src/cpu/nms_cpu.cpp [hunk 5]
```diff
@@ -1,5 +1,7 @@
-// Modified from https://github.com/bharatsingh430/soft-nms/blob/master/lib/nms/cpu_nms.pyx, Soft-NMS is added
-// Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+// Modified from
+// https://github.com/bharatsingh430/soft-nms/blob/master/lib/nms/cpu_nms.pyx,
+// Soft-NMS is added Copyright (c) Facebook, Inc. and its affiliates. All Rights
+// Reserved.
```
Comment (hellock) [inline mmdet/ops/nms/src/cpu/nms_cpu.cpp:4]:
```
// Soft-NMS is added by MMDetection.
// Modified from https://github.com/bharatsingh430/soft-nms/blob/master/lib/nms/cpu_nms.pyx.
// Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
```
